components/Form/index.js

```diff
@@ -66,9 +72,24 @@ export default function Form({ onSubmit }) {
       formObject.image = image;
     }
 
-    localStorage.setItem("favoriteLocation", JSON.stringify(formObject));
+    formObject.id = Date.now().toString();
+
+    const existingLocations =
+      JSON.parse(localStorage.getItem("favoriteLocations")) || []; // Retrieve existing favourite locations or build an empty array
+
+    existingLocations.push(formObject); // Add new favourite location to the existing array
 
-    onSubmit(formObject);
+    localStorage.setItem(
+      "favoriteLocations",
+      JSON.stringify(existingLocations)
+    ); // Save the updated array in local storage
+
+    const newLocationIndex = existingLocations.length - 1;
+    router.push(`/LocationSuccess?index=${newLocationIndex}`);
```

Comment:
This is not how we learned to create dynamic routes. Strictly speaking, your `LocationSuccess` page is not dynamic. The parameter `index` is dynamic, which creates a similar behavior.

Since it works you can leave it as is, but be aware that this is but one way of creating dynamic behavior. Make sure you understand why your solution works.